### ISSUE TITLE
Block Detected Attempting Bypass shows on repeat!

### ISSUE THREAD

**Issue by RediTafa** [futo-org/grayjay-android#1639]
### What happened?

I have been using the app daily and never encountered this issue as everything would always play ,i at worst would need to retry by closing and loading the videos again. 
So today 20/12/2024 i encountered 2 videos that would show a message at the start when it was loading ("Block detected ,attempting bypass") that would repeat its attempts to bypass the block for exactly 4 times before showing a Media error popup that suggested i reload the plugin but that only restarts the loop! (Btw the videos i encountered this are Channel: Asmongold Tv ,Title: CNN just committed treason and Channel: Hits Different Clips ,Title: CDawgVA rizz Japanese woman by calling them young .) Hope this helps narrow the issue down !
![Screenshot_2024-12-21-02-30-31-542_com futo platformplayer-edit](https://github.com/user-attachments/assets/c5dd356d-092e-411b-8b6a-32aa0c2cdc91)

### Grayjay Version

278

### What plugins are you seeing the problem on?

Youtube

### Plugin Version

219

### When do you experience the issue?

- [X] While logged in
- [ ] While logged out
- [ ] N/A

**Comment by Tebrithiel** [comment]
Also experiencing this issue. Workaround: Logging out of the youtube plugin (didn't use clear data option, just in case)

**Comment by RediTafa** [comment]
That does nothing for me.

**Comment by RediTafa** [comment]
It is happening in 3 out of 4 videos i try to watch now so i can't use the app.

**Comment by furdiburd** [comment]
It happens to me too with every single youtube video.

Tried logging out and forcing ump steams (with all combination)

**Comment by Derek-2000** [comment]
I am having this problem as well  and random video will start then stop and restart

**Comment by AlexanderGson** [comment]
Chiming in. I also have the issue with every video just attempting bypass 3 times until that Experimental reload prompt arrives and then it tries the bypass 3 more times over and over.

Haven't been able to find a workaround. So have to use the regular YouTube app atm... It's so much worse to use.

Looking forward to a fix, after the holidays lost likely 😇

**Comment by paridhi089** [comment]
Is this region specific? For me it's running fine without any issues. It an in India. It belive youtube is rolling out some circumvention in certain regions. Maybe everyone should mention their region in the comments.

**Comment by Tebrithiel** [comment]
> Is this region specific? For me it's running fine without any issues. It an in India. It belive youtube is rolling out some circumvention in certain regions. Maybe everyone should mention their region in the comments.

AU/NZ Region. It is a sporadic issue. Sometimes requires closing the entire app to get videos to play again

**Comment by AlexanderGson** [comment]
Northern Europe, Sweden.

Have watched on mobile connection since I first found this thread and commented. Without issues. But on WiFi on both my tablet and on my desktop (also wifi) nothing works what so ever.

**Comment by Pellos21** [comment]
> Northern Europe, Sweden.
> 
> Have watched on mobile connection since I first found this thread and commented. Without issues. But on WiFi on both my tablet and on my desktop (also wifi) nothing works what so ever.

Same

**Comment by furdiburd** [comment]
EU,  currently in Hungary.

On WiFi the videos don't work (no matter where) but with mobile internet it does.

**Comment by Gyttahi** [comment]
Southeastern USA, 
on my end it doesn't seem to be the cell provider for me. Because I'm able to use it on others wifi, I think the providers know who's using the app and sort of just like block them from using it?

**Comment by furdiburd** [comment]
> Southeastern USA, on my end it doesn't seem to be the cell provider for me. Because I'm able to use it on others wifi, I think the providers know who's using the app and sort of just like block them from using it?

Its not getting blocked by your ISP. Its on Google side, but it's probably an IP based block so sometimes switching to a different network (mobilenet, another WiFi etc.) can fix it.

**Comment by ThatNewmanGuy** [comment]
Also seeing the same, UK based; Videos play about 20% and starts to stop/hang.

**Comment by Zvonimir-FUTO** [comment]
Hi everyone, this has been finally fixed and new version of Grayjay and Youtube have been released!

Go update and test and tell us if you find issues except the one below.

We are aware that age restricted, controversial videos, or other videos that require login may not yet work. We will be working on this next week.

The desktop release that is required for this fix will also be released next week.

**Comment by furdiburd** [comment]
I updated the app, but I still have the same issues. Nothing loads and I get a "block detected" message. (YouTube source at v221, app at v281). I've tried enabling force UMP streams, but that doesn't seem to change anything other than the video's playback time.

**Comment by Zvonimir-FUTO** [comment]
> I updated the app, but I still have the same issues. Nothing loads and I get a "block detected" message. (YouTube source at v221, app at v281, manually checked for updates). I've tried enabling force UMP streams, but that doesn't seem to change anything other than the video's playback time.

Can you try disabling **Use Login for video details** if you have enabled it?

**Comment by furdiburd** [comment]
> Can you try disabling **Use Login for video details** if you have enabled it?

It was already off :/

![Image](https://github.com/user-attachments/assets/31c6b14e-130f-4cd0-87b2-b0e414b034e8)

**Comment by Zvonimir-FUTO** [comment]
In the Youtube plugin, at the very bottom, there is a setting called "Show every new botguard token" can you enable it and in toast message you should see start of the token. Could you read the first 5 letters?

**Comment by furdiburd** [comment]
I enabled that and tried to watch YouTube videos in the last 5 minutes but I didn't get any toast message with the token :-/

**Comment by Zvonimir-FUTO** [comment]
> I enabled that and tried to watch YouTube videos in the last 5 minutes but I didn't get any toast message with the token :-/

That is weird. We have one more user claiming the same so you are not alone. It is getting quite late here so will probably have to continue tomorrow.

Do you mind disclosing from which region are you, are you using VPN by chance, are you logged in?

**Comment by Zvonimir-FUTO** [comment]
Other users reported logging out, closing the app and logging in fixed it for them. Could you try that?

**Comment by furdiburd** [comment]
Currently in Hungary, no VPN and yes. I am logged in, will try what you suggested.

**Comment by RediTafa** [comment]
I am the one that opened the issue and there is no issue to be seen anymore as i am using the app!
Thank you !

**Comment by AlexanderGson** [comment]
I've updated the app to version 281 and the youtube plugin is version 221.
Playing videos with WiFi works now, no blocks or anything.

However I've had a few instances of videos stopping at 59 seconds. Sometimes attempted blocks and once a crash.
I did log out of Youtube. Closed Grayjay from memory. Started and logged into youtube plugin.

So perhaps the fix have introduced a couple of other bugs. I'll try it out more tomorrow and get back if I find anything else of value to an eventual troubleshooting.

Not sure if it would be preferable to have a new ticket all together with the new issues, since this one is fixed.

**Comment by RediTafa** [comment]
That issue i had before but not anymore only seen it in the desktop version where it was for every video.

**Comment by mmocnak** [comment]
grayjay app version 281
youtube plugin version 221
region Slovakia
my experience - videos never load on WIFI, I only get black screen with play and pausse buttons shown at the center of the video and 00:00 / -12:55 in time bar.
I've never logged-in to YouTube in Grayjay, I exported my subscriptions from YouTube and imported them to Grayjay
For videos that might be considered controversial (political) I get "Block detected .." . I've enabled "Show every new botguard token" option, forced stop the app, and tried to play video, but no other notification than "Block detected .." was shown.
Interestingly, when I turned off WIFI and tried to play the video - it worked (as before) and the notification with botguard token was shown, but unfortunatelly I didn't catch it and I was not able to reproduce it again by switching between cellular data and WIFI.

**Comment by Zvonimir-FUTO** [comment]
> grayjay app version 281 youtube plugin version 221 region Slovakia my experience - videos never load on WIFI, I only get black screen with play and pausse buttons shown at the center of the video and 00:00 / -12:55 in time bar. I've never logged-in to YouTube in Grayjay, I exported my subscriptions from YouTube and imported them to Grayjay For videos that might be considered controversial (political) I get "Block detected .." . I've enabled "Show every new botguard token" option, forced stop the app, and tried to play video, but no other notification than "Block detected .." was shown. Interestingly, when I turned off WIFI and tried to play the video - it worked (as before) and the notification with botguard token was shown, but unfortunatelly I didn't catch it and I was not able to reproduce it again by switching between cellular data and WIFI.

Can you check if you are getting iOS streams or UMP streams? When you change the video quality it should say which one you have available.

**Comment by mmocnak** [comment]
> Can you check if you are getting iOS streams or UMP streams? When you change the video quality it should say which one you have available.

yes, it's IOS 720p25 video/mp4

**Comment by DickHardly** [comment]
Grayjay version: 281
YouTube plugin Version: 221
Pixel 7 Pro
US East Coast 
Logged out of YouTube 
No forced UMP

Seems I can only get videos to work past 0:59 seconds when the player is using the IOS playback. UMP fall back causes video to black screen and no play at all. After clearing cache and force stopping app a few times videos finally started loading with IOS player back. Sometimes I have to open and close the video a few times for it to finally play with IOS but once it does it works fine.

**Comment by Scoots1994** [comment]
When I start playing a video, if I look at the available download formats if there are no mp4 options the video is going to stop at 59 seconds.  If I stop the video and watch it again (not close the app, clear cache, anything else) if mp4 formats are on the list it's going to play all the way through.

**Comment by Zvonimir-FUTO** [comment]
Seems Youtube is doing some IP banning and labeling some users as bots. We are investigating this.

**Comment by mmocnak** [comment]
> Seems Youtube is doing some IP banning and labeling some users as bots. We are investigating this.

seems like it, yes, but it has to be more sophisticated than just IP banning - I also have Youtube client alternative installed on Android TV (same WIFI network as phone with GrayJay) and everything works on TV (there used to be issues too some months ago, but now everything works again).

**Comment by Zvonimir-FUTO** [comment]
It seems we're all talking about a lot of different issues at this point, and it makes it hard to keep track of whats going on.

Summary:
 - Basic playback appears solved for most users in most situations now.
 - Download has an issue with the new implementation (issue #1849 )
 - Issue with switching networks during playback, which may cause your video playback to stop. (issue #1854 ) 
We are working to fix these issues, but they are unrelated to this particular issue.

Also to repeat, we never disabled iOS source support. Its still prioritized, but they are slowly rolling out that they won't work, and thus Grayjay won't use them if they don't work. Any alignment with the updates is likely because our update was aligned with the wider rollout of this problem (just in time).

**Comment by wuniversales** [comment]
Is it really solved?
I just stopped a video right now with this error, it's weird because in smartube everything works fine...

**Comment by Zvonimir-FUTO** [comment]
> Is it really solved? I just stopped a video right now with this error, it's weird because in smartube everything works fine...

Well Youtube changed something on their backend which broke it again. We have https://github.com/futo-org/grayjay-android/issues/1893 opened for tracking this.
